### DISCUSSION TITLE
is this Outfit Expansion heat affect logical?

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#7756] [General]
Greetings,

I took a bactarian, stripped it completely and started adding new outfits, 4 engines, 5 large shunts, two system cores, other outfits and also installed14 Outfit Expansion.
now it is time to add some weapons, the idle ship heat is at 26.5.
here is the num of Outfit Expansion to idle ship heat ratio:
14 => 26.5
15 => 445.8
16 => 778.3,
17 => 1041
18 => 1247.9
19 => 1410.3
20 => 1537.7
21 => 1637.4
22 => 1715.3
23 => 1776.2
24 => 1823.8
and so on...

I'm not a math genius but I'm failing to see any logic/pattern here...
is this a bug?
I'm running version 0.9.14, my remove bays patchset and this plugin https://github.com/daggs1/Lionheart_Customs

**Comment by mOctave** [comment]
It looks like a polynomial of some sort to me. Note that the equation here is almost certainly not the actual equation being used.
<img width="617" alt="Screenshot 2022-12-02 at 3 56 36 PM" src="https://user-images.githubusercontent.com/73318970/205410741-3e9ad35e-3d70-4804-8720-fc540cb0e40a.png">
Beyond that, @Ferociousfeind, would you like to explain heat?

**Comment by Ferociousfeind** [comment]
This is a complex situation that isn't being fully communicated to you. There is (mostly) sound math and logic going on behind the scenes. 

The largest confounding factor is the already-present outfits. The huge amounts of heat generated, and also the huge amounts of heat destroyed by the heat shunts. You have 5 heat shunts already! 5 large heat shunts gives you -16K heat generation per second, bare.

Then, outfits expansions add "heat inefficiency", which is governed by a complex and opaque formula. The formula has a plateau near 100% strength, where heat inefficiency = 0, and then it drops off towards 0% strength past about 5 heat inefficiency. The curve looks like this:
![image](https://user-images.githubusercontent.com/35383563/205412547-6c58f291-1584-4341-b53f-83a537667c81.png)

0 outfits expansions gives cooling outfits 100% strength. 1 outfits expansion (with 1 cooling inefficiency) gives you 99.62% strength. 5 cooling inefficiency gives your cooling outfits 73.91% strength.

So, at 14 outfits expansions, your 5 heat shunts give you 1,869.91 cooling per second. You said you start with ~26.5 heat per second, which means you have an outfit or a set of outfits producing 1896.5 heat per second (or 31.6 heat per frame) which... I can't find a power generator with mostly exactly that amount of heat (the closest is the wanderer White Sun with 1920 heat per second)

So, the point is, what you're seeing is a stretched and cropped and _flipped_ and out-of-context section of this heat inefficiency curve, the tail end from x=14 and beyond. 

And here, even, here's how it's cropped:
![image](https://user-images.githubusercontent.com/35383563/205413375-bb696aa5-021b-464a-8935-5e43203e6ca9.png)

**Comment by daggs1** [comment]
Thanks for the detailed explanation, can you share the source of this formula please? I'm intrigued...
so bottom line, the calculation comes down to heat inefficiency when it comes to key factor?

**Reply by Ferociousfeind** [reply]
The formula is on line 3373 in source/ship.cpp

**Comment by daggs1** [comment]
found this: https://github.com/endless-sky/endless-sky/blob/4101a9f3ddd298ae59a7dd956e7e71ae1d8ba45f/source/Ship.cpp#L3379 exp stands out there.
I meant the origin of it, not the implementation

**Comment by Ferociousfeind** [comment]
The origin is somewhere in the issues tab on github here. The gist is that "mass expansions" were (apparently) overpowered when used on the Bactrian. They turned it into a heavy but very very strong warship. The solution that MZ came up with was "cooling inefficiency", which reduces the strength of cooling outfits in this way. It is not scientifically driven, only driven by the desire to nerf the bactrian's ability to expand its combat abilities.

**Reply by daggs1** [reply]
thanks for the explanation

**Reply by Zitchas** [reply]
To elborate a touch more: Before the cooling inefficiency was implemented, the vast majority of any ship's cargo space could be directly converted into outfit space. So a Carrier with 820 outfit space and 100 cargo actually basically has 895 outfit and 0 cargo; while a Bulk Freighter with 410 outfit and 600 cargo can actually have 860 outfit space. And the Bactrian, with 740 outfit and 530 cargo turns into a the most impressive ship in human space with over 1100 outfit space.

Basically, if everything converts cargo into outfit space so easily, a) it reduces diversity, and b) one can't have big freighters because even though it makes *sense* to have freighters that can haul vastly more stuff than any ship can install equipment; they could so easily turn that into installation space that freighters risked trivializing warships. and c) it makes it hard to balance, for these reasons.
